lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -463,11 +480,13 @@ function setupEventListeners(transport) {
     if (transport._twilioConnection) {
       transport._twilioConnection.removeListener('message', handleMessage);
     }
-    const { _options, _wsServer } = transport;
+    const { _iceServersStatus, _options, _wsServer, state } = transport;
     const { TwilioConnection } = _options;
 
     const twilioConnection = new TwilioConnection(_wsServer, Object.assign({
-      helloBody: transport._createConnectOrSyncOrDisconnectMessage()
+      helloBody: state === 'connecting' && _iceServersStatus === 'acquire'
+        ? transport._createIceMessage()
```

Comment:
we want to get ice servers just once, and not for every reconnection right? Wouldnt this cause ice message to be sent for every reconnection ?

Reply:
@makarandp0 Nope. During reconnects, the `state` is in `reconnecting`, so the hello body will now contain a "sync" message.